src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -10,502 +11,314 @@
-                Assert.Null(obj.C.J.A);
-                Assert.Equal(7, obj.C.J.B);
+                await Assert.ThrowsAsync<JsonException>(async () => await JsonSerializer.DeserializeAsync(stream, type, readOptions));
             }
         }
 
         [Theory]
-        [MemberData(nameof(TestData))]
-        public static async Task ContinuationShouldWorkAtAnyPosition_ValueType_ValueType(int paddingLength, bool ignoreNullValues)
+        [MemberData(nameof(TestData), /* enumeratePayloadTweaks: */ false)]
+        public static void ShouldWorkAtAnyPosition_Sequence(string json, int bufferSize, Type type, bool ignoreNullValues)
         {
-            var stream = new MemoryStream();
-            {
-                var obj = new Outer<TestValueType<NestedValueType>>
-                {
-                    S = new string('x', paddingLength),
-                    C = new()
-                    {
-                        A = "Hello",
-                        B = null,
-                        C = 42,
-                        D = null,
-                        E = 3.14e+17f,
-                        F = null,
-                        G = true,
-                        H = null,
-                        I = new int[] { 42, 17 },
-                        J = new()
-                        {
-                            A = null,
-                            B = 7,
-                        }
-                    }
-                };
-                await JsonSerializer.SerializeAsync(stream, obj, new JsonSerializerOptions { Converters = { new OuterConverter<TestClass<NestedClass>>() } });
-            }
+            var readOptions = new JsonSerializerOptions { IgnoreNullValues = ignoreNullValues, };
 
-            stream.Position = 0;
-            {
-                var readOptions = new JsonSerializerOptions
-                {
-                    DefaultBufferSize = 128,
-                    IgnoreNullValues = ignoreNullValues,
-                };
+            var chunk = new Chunk(json, bufferSize);
+            var sequence = new ReadOnlySequence<byte>(chunk, 0, chunk.Next, chunk.Next.Memory.Length);
 
-                Outer<TestValueType<NestedValueType>> obj = await JsonSerializer.DeserializeAsync<Outer<TestValueType<NestedValueType>>>(stream, readOptions);
-
-                Assert.Equal(new string('x', paddingLength), obj.S);
-                Assert.Equal("Hello", obj.C.A);
-                Assert.Null(obj.C.B);
-                Assert.Equal(42, obj.C.C);
-                Assert.Null(obj.C.D);
-                Assert.Equal(3.14e17f, obj.C.E);
-                Assert.Null(obj.C.F);
-                Assert.True(obj.C.G);
-                Assert.Null(obj.C.H);
-                Assert.Collection(obj.C.I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
-                Assert.Null(obj.C.J.A);
-                Assert.Equal(7, obj.C.J.B);
-            }
+            var reader = new Utf8JsonReader(sequence, true, default);
```

Comment:
nit: Simplify the ctor since we aren't testing partial data/re-entrancy:
```suggestion
            var reader = new Utf8JsonReader(sequence);
```

Alternatively, use named parameters.